### PR TITLE
Set-AzVMOperatingSystem AssessmentMode parameter

### PR DESCRIPTION
**Pull request by Sandido**
## Description

Adding the new AssessmentMode parameter to the Set-AzVMOperatingSystem.

Design spec: https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/940

## Checklist

- [ ] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/Compute/Compute/VirtualMachine/Config/SetAzureVMOperatingSystemCommand.cs [hunk 1]
```diff
@@ -279,6 +279,28 @@ public class SetAzureVMOperatingSystemCommand : Microsoft.Azure.Commands.Resourc
...
+            HelpMessage = "Automatic assessment mode value for the virtual machine. Possible values are ImageDefault and AutomaticByPlatform.")]
+        [Parameter(
+            ParameterSetName = WindowsDisableVMAgentWinRmHttpsParamSet,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Automatic assessment mode value for the virtual machine. Possible values are ImageDefault and AutomaticByPlatform.")]
+        [Parameter(
+            ParameterSetName = LinuxParamSet,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Automatic assessment mode value for the virtual machine. Possible values are ImageDefault and AutomaticByPlatform.")]
+        public string AssessmentMode { get; set; }
```
Comment (BethanyZhou) [inline src/Compute/Compute/VirtualMachine/Config/SetAzureVMOperatingSystemCommand.cs:302]: Recommend to add a PSArgumentCompleter for this parameter.

**Review by BethanyZhou — CHANGES_REQUESTED** [review]

File: src/Compute/Compute/VirtualMachine/Config/SetAzureVMOperatingSystemCommand.cs (hunk 1 above)
Comment (Sandido) [reply src/Compute/Compute/VirtualMachine/Config/SetAzureVMOperatingSystemCommand.cs:303]: Added

**Comment by Sandido** [comment]
@BethanyZhou , This is ready for review and merging.

**Review by BethanyZhou — APPROVED** [review]
LGTM

**Pull request merged**